genesyscloud/architect_grammar_language/resource_genesyscloud_architect_grammar_language_utils.go

```diff
@@ -0,0 +1,163 @@
+	} else if fileType == "dtmf" {
+		if dtmfData := d.Get("dtmf_file_data").([]interface{}); len(dtmfData) > 0 {
+			dtmfDataMap := dtmfData[0].(map[string]interface{})
+			if hash, ok := dtmfDataMap["file_content_hash"].(string); ok {
+				metadataMap["file_content_hash"] = hash
+			}
+		}
+	}
+
+	return []interface{}{metadataMap}
+}
+
+func ArchitectGrammarLanguageResolver(languageId, exportDirectory, subDirectory string, configMap map[string]interface{}, meta interface{}) error {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := getArchitectGrammarLanguageProxy(sdkConfig)
+
+	fullPath := path.Join(exportDirectory, subDirectory)
+	if err := os.MkdirAll(fullPath, os.ModePerm); err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+	grammarId, languageCode := splitLanguageId(languageId)
+	language, _, err := proxy.getArchitectGrammarLanguageById(ctx, grammarId, languageCode)
+	if err != nil {
+		return err
+	}
+
+	if language.VoiceFileMetadata != nil && language.VoiceFileUrl != nil {
+		if language.VoiceFileMetadata.FileType != nil {
+			downloadFiles(grammarId, fullPath, *language.Language, *language.VoiceFileUrl, *language.VoiceFileMetadata.FileType)
+		} else {
+			downloadFiles(grammarId, fullPath, *language.Language, *language.VoiceFileUrl, "")
+		}
+	}
+
+	if language.DtmfFileMetadata != nil && language.DtmfFileUrl != nil {
+		if language.DtmfFileMetadata.FileType != nil {
+			downloadFiles(grammarId, fullPath, *language.Language, *language.DtmfFileUrl, *language.DtmfFileMetadata.FileType)
+		} else {
+			downloadFiles(grammarId, fullPath, *language.Language, *language.DtmfFileUrl, "")
+		}
+	}
+
+	updateFilenamesInExportConfigMap(configMap, grammarId, *language, subDirectory)
+	return nil
+}
+
+func downloadFiles(grammarId string, fullPath string, languageCode string, fileUrl string, fileTypeSdk string) error {
+	fileType := ""
+	if fileTypeSdk != "" {
+		fileType = strings.ToLower(fileTypeSdk)
+	}
+	dtmfFileName := fmt.Sprintf("%s-dtmf-%s.%s", languageCode, grammarId, fileType)
+	if err := files.DownloadExportFile(fullPath, dtmfFileName, fileUrl); err != nil {
+		return err
+	}
+	return nil
+}
+
+func updateFilenamesInExportConfigMap(configMap map[string]interface{}, grammarId string, language platformclientv2.Grammarlanguage, subDir string) {
+	if voiceFileData, ok := configMap["voice_file_data"].([]interface{}); ok {
+		setExporterFileData(voiceFileData, grammarId, language, subDir, "voice")
+	}
+	if dtmfFileData, ok := configMap["dtmf_file_data"].([]interface{}); ok {
+		setExporterFileData(dtmfFileData, grammarId, language, subDir, "dtmf")
+	}
+}
+
+func setExporterFileData(fileDataMap []interface{}, grammarId string, language platformclientv2.Grammarlanguage, subDir string, fileType string) {
+	//Set file name and content hash in the exporter map
+	if fileData, ok := fileDataMap[0].(map[string]interface{}); ok {
+		fileExtension := ""
+		if fileType == "voice" && language.VoiceFileMetadata.FileType != nil {
```

Comment:
I see magic strings for voice and dtmf in a couple of different spots.  I would extract these out into constant values and reference the constants instead of the direct strings.